### PR TITLE
Add SPDX License information

### PR DESCRIPTION
**Pull request by jonasstein**
Adding license information according to the SPDX stanard.
Used the reuse tool.

### CODE REVIEW

File: tests/catch.hpp [hunk 1]
```diff
@@ -1,13 +1,8 @@
...
- *  This file has been merged from multiple headers. Please don't edit it directly
- *  Copyright (c) 2022 Two Blue Cubes Ltd. All rights reserved.
- *
- *  Distributed under the Boost Software License, Version 1.0. (See accompanying
- *  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
- */
+// Copyright (c) 2022 Two Blue Cubes Ltd. All rights reserved.
+// SPDX-FileCopyrightText: 2022 stenc authors
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
```
Comment (sunwire) [inline tests/catch.hpp:4]: Are you sure about this? I think it should be BSL-1.0

Comment (jonasstein) [reply tests/catch.hpp]: definitively. Fixed that. Thank you.

**Review by sunwire — APPROVED** [review]
I think we should also delete the [COPYING](https://github.com/scsitape/stenc/blob/master/COPYING) file, because it will be replaced by the LICENSES/GPL-2.0-or-later.txt file.

**Comment by jonasstein** [comment]
agreed. I removed COPYING

**Pull request merged**